src/components/Quotes.js

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from 'react';
+
+const category = 'happiness';
+
+function Quotes() {
+  const [data, setData] = useState([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      try {
+        const res = await fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`,
+          {
+            headers: {
+              'X-Api-Key': 'IJxDQE0w1qFvps1T42CbGw==2TlSWBd5MFRPAT17',
+              'Content-Type': 'application/json',
+            },
+          });
+        const json = await res.json();
+        setData(json[0]);
+      } catch (error) {
+        setHasError(true);
+      }
+      setIsLoading(false);
+    };
+    fetchData();
+  }, [setData, setIsLoading]);
+
```

Comment:
- [ ] Good job so far 👍👍A quote is being displayed with the correct API key, but when I am using a wrong API key, following error is occurring instead of an error message ⚠ Kindly fix this and make sure to handle the error correctly. You can consider using the following 👇 Please test it at your end before asking for a review

![image](https://github.com/Ruthmy/math-magicians/assets/106119105/60440f11-f1d5-402c-aea2-55c80cc36057)

```
useEffect(() => {
  const fetchData = async () => {
    setIsLoading(true);
    try {
      const res = await fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`,
        {
          headers: {
            'X-Api-Key': 'IJxDQE0w1qFvps1T42CbGw==2TlSWBd5MFRPAT17',
            'Content-Type': 'application/json',
          },
        });
      if (res.ok) {
        const json = await res.json();
        setData(json[0]);
      } else {
        setHasError(true);
      }
    } catch (error) {
      setHasError(true);
    } finally {
      setIsLoading(false);
    }
  };
  fetchData();
}, []);

```